tests/unit/test_contracts/explicit_nested_tests.cpp

```diff
@@ -32,7 +32,7 @@ CONTRACT explicit_nested_tests : public contract {
       std::tuple<uint64_t, std::optional<float>, std::vector<int>> tup1;
       std::variant<uint64_t, std::optional<std::pair<int, float>>, std::vector<int>> var1;
       std::vector<std::vector<_mystruct>> vvmys;
-      auto primary_key() const { return id; }
+      auto primary_key() const -> decltype(id) { return id; }
```

Comment:
if that is needed for every case we may break many contracts. Are you sure we need this?

Reply:
I honestly have no idea why this is necessary, but in multiple environments I and another dev see a build failure here unless the hint is provided. I am quite confident that standard C++ does not require the use of the hint here, but the peculiar Clang used to build contracts seems to get confused in this particular spot for some reason.